qiita_pet/handlers/api_proxy/processing.py

```diff
@@ -104,14 +117,27 @@ def _helper_process_params(params):
                 del params[k]
             eparams.append(_helper_process_params(params))
 
+        # removing this artifact from extra_artifacts
+        if artifact.artifact_type in extra_atypes:
+            extra_atypes.remove(artifact.artifact_type)
+
+    extra_artifacts = defaultdict(list)
```

Comment:
@antgonza Just to confirm, you're using a defaultdict() here, but do you set the default value w/setdefault() anywhere?

Reply:
no need to initialize, that's the point of a default dict, see line 130.

Reply:
Ah I see. when 'list' is passed as the parameter to defaultdict(), the default is set to a sensible empty list, much like 'int' will set the default to 0.